plugins/inputs/suricata/suricata_test.go

```diff
@@ -0,0 +1,331 @@
+
+	c, err := net.Dial("unix", tmpfn)
+	if err != nil {
+		log.Println(err)
+	}
+	c.Write([]byte(ex4))
+	c.Write([]byte("\n"))
+	c.Close()
+
+	acc.Wait(1)
+	acc.AssertContainsTaggedFields(t, "suricata",
+		map[string]interface{}{
+			"capture.kernel_packets": float64(905344474),
+			"capture.kernel_drops":   float64(78355440),
+		},
+		map[string]string{"thread": "W1#en..bar1"})
+
+	s.Stop()
+}
+
+func TestSuricataInvalidPath(t *testing.T) {
+	tmpfn := fmt.Sprintf("/t%d/X", rand.Int63())
+
+	s := Suricata{
+		Source: tmpfn,
+	}
+
+	acc := testutil.Accumulator{}
+	acc.SetDebug(true)
+
+	assert.Error(t, s.Start(&acc))
+}
+
+func TestSuricataTooLongLine(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+	tmpfn := filepath.Join(dir, fmt.Sprintf("t%d", rand.Int63()))
+
+	s := Suricata{
+		Source: tmpfn,
+	}
+	acc := testutil.Accumulator{}
+	acc.SetDebug(true)
+
+	assert.NoError(t, s.Start(&acc))
+
+	c, err := net.Dial("unix", tmpfn)
+	if err != nil {
+		log.Println(err)
+	}
+	c.Write([]byte(strings.Repeat("X", 20000000)))
+	c.Write([]byte("\n"))
+	c.Close()
+
+	acc.WaitError(1)
+
+	s.Stop()
+}
+
+func TestSuricataDisconnectSocket(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+	tmpfn := filepath.Join(dir, fmt.Sprintf("t%d", rand.Int63()))
+
+	s := Suricata{
+		Source: tmpfn,
+	}
+	acc := testutil.Accumulator{}
+	acc.SetDebug(true)
+
+	assert.NoError(t, s.Start(&acc))
+
+	c, err := net.Dial("unix", tmpfn)
+	if err != nil {
+		log.Println(err)
+	}
+	c.Write([]byte(ex2))
+	c.Write([]byte("\n"))
+	c.Close()
+
+	c, err = net.Dial("unix", tmpfn)
+	if err != nil {
+		log.Println(err)
+	}
+	c.Write([]byte(ex3))
+	c.Write([]byte("\n"))
+	c.Close()
+
+	acc.Wait(2)
+
+	s.Stop()
+}
+
+func TestSuricataPluginDesc(t *testing.T) {
+	v, ok := inputs.Inputs["suricata"]
+	if !ok {
+		t.Fatal("suricata plugin not registered")
+	}
+	desc := v().Description()
+	if desc != "Suricata stats plugin" {
+		t.Fatal("invalid description ", desc)
+	}
+}
+
+func TestSuricataStartStop(t *testing.T) {
+	v, ok := inputs.Inputs["suricata"]
+	if !ok {
+		t.Fatal("suricata plugin not registered")
+	}
+	s := v().(telegraf.ServiceInput)
+	acc := testutil.Accumulator{}
+	acc.SetDebug(true)
+	assert.NoError(t, s.Start(&acc))
+	s.Stop()
+}
+
+func TestSuricataDoubleStart(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+	tmpfn := filepath.Join(dir, fmt.Sprintf("t%d", rand.Int63()))
+
+	s := Suricata{
+		Source: tmpfn,
+	}
+	acc := testutil.Accumulator{}
+	acc.SetDebug(true)
+	assert.NoError(t, s.Start(&acc))
+	assert.NoError(t, s.Start(&acc))
+
+	c, err := net.Dial("unix", tmpfn)
+	if err != nil {
+		log.Println(err)
+	}
+	c.Write([]byte(ex2))
+	c.Write([]byte("\n"))
+	c.Close()
+
+	acc.Wait(1)
+	s.Stop()
+}
+
+func TestSuricataDoubleStop(t *testing.T) {
```

Comment:
Same, no need to test as this won't happen.

Reply:
Removed.